README.md

```diff
@@ -117,16 +117,73 @@ Ninja:
 
     utils/build-script -r -t
 
-## Developing Swift in Xcode
+### Development Environment
 
-`build-script` can also generate Xcode projects:
+There are two primary build options to use with CMake, xcodebuild or Ninja.
+xcodebuild integrates nicely with Xcode.app, but Ninja is a bit faster and
+supports more environments.
+
+To build using Xcode, run:
 
     utils/build-script -x
 
 The Xcode IDE can be used to edit the Swift source code, but it is not currently
 fully supported as a build environment for SDKs other than macOS. If you need to
 work with other SDKs, you'll need to create a second build using Ninja.
 
+To build using Ninja, run:
+
+    utils/build-script -R --debug-swift
+
+This will build LLVM+Clang with optimizations and no debug symbols, but will
+build swift with debug symbols. This is a good starting point since the debugger
+will work in the swift code base, and the tests will run faster.
+
+#### Build Products
+
+All of the build products are placed in `swift-source/build/${TOOL}-${MODE}/${PRODUCT}-${PLATFORM}/`.
+If macOS swift with Ninja in DebugAssert mode was built, all of the products
+would be in `swift-source/build/Ninja-DebugAssert/swift-macosx-x86_64/`. It
+helps to save this directory as an environment variable for future use.
+
+    export SWIFT_BUILD_ROOT="~/swift-source/build/Ninja-DebugAssert/swift-macosx-x86_64"
+
+#### Ninja
+
+Once the first build has been finished with `build-script`, ninja can perform
+fast incremental builds of various products.
+
+    cd ${SWIFT_BUILD_ROOT}
+    ninja swift
+    ninja swift-stdlib
+
+These incremental builds are a big timesaver when developing and debugging. It
+is still always a good idea to do a full build after using `update-checkout`.
+Building the `swift-stdlib` target as an additional layer of testing from time
+to time is also a good idea.
+
+#### Xcode
+
+To open the swift in Xcode, open `${SWIFT_BUILD_ROOT}/Swift.xcodeproj`. It will
+auto-create a *lot* of schemes for all of the available targets. A common debug
+flow would involve:
+
+ - Select the ‘swift’ scheme
+ - Pull up the scheme editor (⌘+⇧+<)
+ - Select the ‘Arguments’ tab and click the ‘+’
+ - Add the command line options
+ - Close the scheme editor
+ - Build and run
```

Comment:
I stole this workflow from a recent swift-dev article, and I think including both is a good idea. Thanks!